### PR TITLE
Enable blobless fetch for git repos

### PR DESCRIPTION
**Pull request by howardjohn**
This should be quite a bit faster

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ plank:
       gcs_configuration:
         bucket: "istio-prow"
         path_strategy: "explicit"
+      blobless_fetch: true
```
Comment (zirain) [inline prow/config.yaml:21]: can you details about this?

Comment (howardjohn) [reply prow/config.yaml:21]: https://github.com/kubernetes/test-infra/issues/26590

tl;dr `git clone` faster, should have zero downsides I think (or if I am wrong we can just revert)

**Review by linsun — COMMENTED** [review]
cc @dhawton who is our prow expert to review

File: prow/config.yaml (hunk 1 above)
Comment (zirain) [reply prow/config.yaml:21]: TIL, thanks!

**Review by zirain — APPROVED** [review]

**Review by dhawton — APPROVED** [review]

**Review by linsun — APPROVED** [review]

**Review by kfaseela — APPROVED** [review]

**Comment by ericvn** [comment]
@stewartbutler Would you take a look?

**Review by stewartbutler — APPROVED** [review]

**Pull request merged**